rbac/common/base/base_message.py

```diff
@@ -225,28 +362,187 @@ def get(self, object_id, target_id=None):
+
+    def _get_store(self, object_id, target_id, outputs, output_state):
+        """Gets the store object to store data for this message"""
+        address = self.address(object_id=object_id, target_id=target_id)
+        container = None
+        if address not in outputs and address in output_state:
+            raise ValueError(
+                "Address {} not in listed outputs".format(addresser.parse(address))
+            )
+        if address in output_state:
+            container = output_state[address]
+            # TODO: is getting the first item in the container... may not be correct!
+            store = getattr(container, self._state_container_list_name)[0]
+        if not container:
+            container, store = self._get_new_state()
+            output_state[address] = container
+        return store
+
+    def _update_store(
+        self, object_id, target_id, message, store, outputs, output_state, signer
+    ):
+        """Update the output state entries for the given address using
+        the data the given message"""
         address = self.address(object_id=object_id, target_id=target_id)
-        state_client.set_address(state=state, address=address, container=container)
+        self.store_message(
+            object_id=object_id,
+            target_id=target_id,
+            store=store,
+            message=message,
+            outputs=outputs,
+            output_state=output_state,
+            signer=signer,
+        )
+        output_state["changed"].add(address)
 
-    def apply(self, header, payload, state):
+    # pylint: disable=unused-argument
+    def store_message(
+        self, object_id, target_id, store, message, outputs, output_state, signer
+    ):
+        """Copies a message to the state object, excluding properties
+        listed in message_fields_not_in_state property. This provides
+        a simple default behavior for cases where this is appropriate;
+        commonly override this method in message classes"""
+        batcher.message_to_message(
+            message_to=store,
+            message_from=message,
+            message_name=self._name_camel,
+            exclude_fields=self.message_fields_not_in_state,
+        )
+
+    def save_state(self, context, output_state):
+        """Save the output state to the blockchain"""
+        changed = [
+            address
+            for address in output_state.keys()
+            if address in output_state["changed"]
+        ]
+        entries = {}
+        for address in changed:
+            entries[address] = output_state[address].SerializeToString()
+        state_client.set_state(context=context, entries=entries)
+
+    def apply_update(
+        self, message, object_id, target_id, outputs, output_state, signer
+    ):
+        """Apply additional state changes (if any)
+        Message implementation will override this method if there
+        is data to be stored in any other address"""
+        pass
```

Comment:
Why is this function just to pass?

Reply:
Because you can override it but aren't required to. It's a hook if a message needs to store data in other addresses.